text/0000-gps_support.md

```diff
@@ -0,0 +1,24 @@
+# NavSatFix (GPS) Support in Cartographer ROS
+
+## Summary
+[summary]: #summary
+
+The goal is to add NavSatFix (GPS) support to Cartographer ROS. 
+The NavSatFix (GPS) data will be used in pose graph optimization, using the fixed frame pose feature of Cartographer.
+
+## Motivation
+[motivation]: #motivation
+
+In outdoor applications, especially cars, GPS data is commonly available. 
+Using this data will reduce the error in location, especially over large distances. 
+
+## Approach
+[approach]: #approach
+
+We will wire up the [sensor_msgs/NavSatFix](http://docs.ros.org/api/sensor_msgs/html/msg/NavSatFix.html) and introduce two new options for this: `use_nav_sat` and `nav_sat_sampling_ratio`.
```

Comment:
Why do you want to feed the fixed-frame poses from a GPS specific message type? Maybe other people want to use a different message type. Aren't fixed-frame poses just time-stamped poses? Why not use a ROS msg which contains just that?

Reply:
`NavSatFix` is the message widely used in the ROS community, e.g. what a GPS driver provides. This is similar to the code we already have in Cartographer which understands `LaserScan` messages and translates them to the Cartographer-specific data format. `NavSatFix` is not GPS specific and supports other GNSS systems.

Reply:
I see, but it is specific to satellite based positioning systems. I see the `NavSatStatus` submessage in there. Why not simply specify the fixed-frame pose as timed `geometry_msgs/Pose.msg`?

Most people use Cartographer for indoor robot navigation. In this setup it is very unlikely that your fixed-frame is coming from a satellite based system. Don't you think it is unfortunate that everyone now has to invent a `NavSatStatus` that is completely ignored  by your conversion code?

I am not opposed to using `NavSatFix` messages. I just want to make sure we  implement this feature in a way that it appeals to the general usecase.

Reply:
I think it also makes sense to add a message that allows specifying fixed frame poses directly for other uses cases. To me this seems similar how Cartographer ROS already support 3 ways to provide range data.

This RFC is about adding GPS-like data. Providing a generic message instead of `NavSatFix`would be cumbersome (everyone would have to convert to a cartesian coordinate system) and as mentioned having a `/fix`topic is common in ROS.

Would you agree that supporting `NavSatFix` and a generic fixed frame pose message are two orthogonal improvements to Cartographer ROS?

Reply:
The drawback of this is that it requires the introduction of another `use_XXX` and `XXX_sampling_ratio`. Or I guess if you would use a more generic name than `use_nav_sat` we could use message introspection to handle both types of messages by the same option.

Reply:
Shouldn't we keep this consistent with the way different `LaserScan`, `MultiEchoLaserScan` and `PointCloud2`. Also, the topic `/fix` is very specific to `NavSatFix`. We can rename the sampling ratio to `fixed_frame_pose_sampling_ratio` if you would like this to be forward looking to when a different way to handle fixed frame poses is added. Otherwise we would refactor it then.

Reply:
Sounds good, let's try to share the sampler then at least.